lib/serverless/aws-lambda.js

```diff
@@ -251,6 +278,17 @@ class AwsLambda {
     }
   }
 
+  wrapStreamAndCaptureError(transaction, txnEnder, stream) {
+    const shim = this.shim
+    stream.on('error', (error) => {
+      shim.agent.errors.add(transaction, error)
+    })
+    stream.on('end', () => {
```

Comment:
i also think this event should be `finish` based on the docs. 

```
Event: 'finish'[#](https://nodejs.org/api/stream.html#event-finish)
Added in: v0.9.4
The 'finish' event is emitted after the [stream.end()](https://nodejs.org/api/stream.html#writableendchunk-encoding-callback) method has been called, and all data has been flushed to the underlying system.
```

Reply:
Toggling to `finish` works, and reveals four test failures: 
* should record error event when func is async an UnhandledPromiseRejection is thrown
* should record error event when error is thrown
* should not end transactions twice
* should record standard background metrics

Reply:
That's because they're ending the streaming before an error exists. If you remove ending the stream it works.